### DISCUSSION TITLE
Average Yield Over Time with Position Updates (DeFi Use Case)

### DISCUSSION THREAD

**Discussion by luluwebmaster** [ghostfolio/ghostfolio#4046] [Ideas]
Hello, I’d like to suggest a feature that calculates the average yield of a position based on updated values over time, with a particular focus on decentralized finance (DeFi) scenarios. The goal is to allow users to input multiple updates for a position’s value (e.g., a liquidity position or loan balance on platforms like Aave), and Ghostfolio would calculate the yield between each update date. This would enable users to track the growth of their DeFi investments dynamically over time, reflecting real-time market conditions and reinvestment outcomes.

### **Use Case Example**
1. **Initial Position Update**:  
   - Date: 01-01-2024  
   - Value: $100 (e.g., staked in an Aave lending pool)

2. **Position Update**:  
   - Date: 01-02-2024  
   - Updated Value: $110 (due to accrued interest or additional deposits)  
   - **Yield Calculation**: Compute the yield from the initial date (01-01-2024) to this update date (01-02-2024).

3. **Subsequent Updates**:  
   - Each time the user updates the position’s value (for example, after withdrawing rewards or reinvesting earnings on Aave), calculate the yield between the previous update and the new update. For instance, if a new update on 01-03-2024 shows a value of $120, calculate the yield between 01-02-2024 and 01-03-2024.
   - This calculation should ideally support an unlimited number of updates, as DeFi positions are frequently modified with deposits, withdrawals, and interest accrual.

### **Suggested Implementation**
- **User Interface**: Provide an option for users to update their position values on specific dates.
- **Backend Calculations**: Implement a function to calculate and store the yield percentage between each consecutive update date.
- **Display**: Show the cumulative yield and/or the average yield on the user’s portfolio page. This would provide a transparent view of performance over time for each DeFi position.

### **DeFi-Specific Considerations**
- **Platform Compatibility**: Many DeFi platforms like Aave, Compound, and others accrue interest or rewards continuously. This feature could greatly benefit users by offering a straightforward way to monitor these incremental gains.
- **Graphical Representation**: Display yield progression over time through a graph to help users visualize growth across all updates, potentially highlighting market conditions or reinvestment impacts.

Thank you for considering this feature! It would significantly enhance Ghostfolio’s utility for DeFi investors who seek a clear understanding of yield progression on variable positions.

---

**Note**: If you're interested in this feature, please give this post a 👍 to show your support!

**Comment by GerardPolloRebozado** [comment]
any updates? how many people have to be interested for this to be a feature?
